### PR TITLE
Added some unit tests

### PR DESCRIPTION
**Pull request by mrzenioszeniou**
More of a way to explore the codebase rather than introduce useful tests really.

### CODE REVIEW

File: chain-impl-mockchain/src/testing/gen/mod.rs [hunk 1]
```diff
@@ -102,4 +105,19 @@ impl TestGen {
...
+        let leader_pair = TestGen::leader_pair();
+        let header_id = TestGen::hash();
+        let mut ie = ConfigParams::new();
+        ie.push(ConfigParam::Discrimination(Discrimination::Test));
+        ie.push(ConfigParam::AddBftLeader(leader_pair.leader_id));
+        ie.push(ConfigParam::SlotDuration(10u8));
+        ie.push(ConfigParam::SlotsPerEpoch(10u32));
+        ie.push(ConfigParam::KesUpdateSpeed(3600));
+        ie.push(ConfigParam::Block0Date(crate::config::Block0Date(0)));
+        ie.push(ConfigParam::Block0Date(crate::config::Block0Date(0)));
```
Comment (dkijania) [inline chain-impl-mockchain/src/testing/gen/mod.rs:119]: duplicates line above

File: chain-impl-mockchain/src/testing/gen/mod.rs [hunk 2]
```diff
@@ -102,4 +105,19 @@ impl TestGen {
             })
             .build()
     }
+
+    pub fn ledger() -> Ledger {
+        let leader_pair = TestGen::leader_pair();
+        let header_id = TestGen::hash();
+        let mut ie = ConfigParams::new();
+        ie.push(ConfigParam::Discrimination(Discrimination::Test));
```
Comment (dkijania) [inline chain-impl-mockchain/src/testing/gen/mod.rs:113]: those values are not truly random, but we can live with it for the moment. I imagine in the future we will randomize some of the settings like slot_duration or slots_per_epoch.

File: chain-crypto/src/algorithms/vrf/mod.rs [hunk 3]
```diff
@@ -84,3 +84,20 @@ impl VerifiableRandomFunction for RistrettoGroup2HashDh {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    /// `secret_from_binary` should fail if the provided byte array does not match the public key size
+    fn vtf_size_checks() {
+        for n in 0..u8::MAX as usize {
+            let secret_key = RistrettoGroup2HashDh::secret_from_binary(&vec![0; n]);
```
Comment (dkijania) [inline chain-crypto/src/algorithms/vrf/mod.rs:96]: as discussed, let's try to rewrite test so it uses arbitrary for size of input vector

File: chain-crypto/src/algorithms/sumed25519/mod.rs [hunk 4]
```diff
@@ -115,3 +115,30 @@ impl KeyEvolvingSignatureAlgorithm for SumEd25519_12 {
...
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    /// `public_from_binary` and `signature_from_bytes` should fail if the provided byte array does
+    /// not match the public key size
+    fn sum_ed25519_12_size_checks() {
+        for n in 0..u8::MAX as usize {
```
Comment (dkijania) [inline chain-crypto/src/algorithms/sumed25519/mod.rs:127]: as discussed, let's try to rewrite test so it uses arbitrary for size of input vector

**Review by dkijania — CHANGES_REQUESTED** [review]

File: chain-impl-mockchain/src/testing/gen/mod.rs (hunk 1 above)
Comment (mrzenioszeniou) [reply chain-impl-mockchain/src/testing/gen/mod.rs]: Gone

File: chain-impl-mockchain/src/testing/gen/mod.rs (hunk 2 above)
Comment (mrzenioszeniou) [reply chain-impl-mockchain/src/testing/gen/mod.rs:114]: I left a TODO behind for the time being.

File: chain-crypto/src/algorithms/vrf/mod.rs (hunk 3 above)
Comment (mrzenioszeniou) [reply chain-crypto/src/algorithms/vrf/mod.rs]: Done. I also changed the test slightly to ensure we don't have any false `SizeInvalid` instances.

File: chain-crypto/src/algorithms/sumed25519/mod.rs (hunk 4 above)
Comment (mrzenioszeniou) [reply chain-crypto/src/algorithms/sumed25519/mod.rs]: Done.

**Review by dkijania — APPROVED** [review]

**Comment by dkijania** [comment]
very nice!

**Pull request merged**